Block/Analytics.php

```diff
@@ -61,17 +40,35 @@ public function getEvents()
         );
     }
 
-    /**
-     * Render metrilo js if module is enabled
-     *
-     * @return string
-     * @codeCoverageIgnore
-     */
     protected function _toHtml()
     {
         if (!$this->helper->isEnabled($this->helper->getStoreId())) {
             return '';
         }
         return parent::_toHtml();
     }
+
+    public function getEvent()
+    {
+        if (!$this->fullActionName || $this->isRejected($this->fullActionName)) {
+            return;
+        }
+        
+        switch($this->fullActionName) {
+            // product view pages
+            case 'catalog_product_view':
+                return new \Metrilo\Analytics\Model\Events\ProductViewEvent($this->coreRegistry->registry('current_product')->getId());
```

Comment:
You're spreading the knowledge of how to build the event in here. All the knowledge should be in the `ProductViewEvent` class. What do you think about passing the registry and letting the ProductViewEvent know what to take out from it?

Reply:
It sounds more reasonable, will fix it.